### ISSUE TITLE
[1.x] `previousCompile` triggers upstream `compile`

### ISSUE THREAD

**Issue by Friendseeker** [sbt/sbt#7982]
## steps

- Checkout https://github.com/Friendseeker/sbt-7973/tree/previousCompile-trigger-compile
- Call `B / previousCompile`

## problem

A compilation is triggered
```
[IJ]B / previousCompile
[debug] Other repositories:
[debug] Default repositories:
[debug] Using inline dependencies specified in Scala.
[debug] Copy resource mappings:
[debug]
[debug] [zinc] IncrementalCompile -----------
[debug] IncrementalCompile.incrementalCompile
[debug] previous = Stamps for: 3 products, 1 sources, 2 libraries
[debug] current source = Set(${BASE}/A/src/main/scala/A.scala)
```

## expectation

`previousCompile` should merely return the result of previous compilation. It should never trigger compilation of upstream project.

## note

Here's the trace

- previousCompile
- compileIncSetup
- dependencyPicklePath
- dependencyVirtualClasspath
- dependencyClasspath
- (concat)
- internalDependencyClasspath
- (internalDependencyClasspathTask)
- (internalDependenciesImplTask)
- exportedProducts
- (trackedExportedProducts)
- (trackedExportedProductsImplTask)
- (trackedNonJarProductsImplTask)
- compile

Basically `compileIncSetup` gathers a list of upstream analysis, which I guess do... kinda make sense. The issue is probably that `previousCompile` should not call `compileIncSetup` in the first place.
